quickjs.c

```diff
@@ -6673,26 +6675,44 @@ static void build_backtrace(JSContext *ctx, JSValue error_val, JSValue filter_fu
     JSRuntime *rt;
     JSCallSiteData csd[64];
     uint32_t i;
+    double d_stack_trace_limit;
     int stack_trace_limit;
 
+    rt = ctx->rt;
+    if (rt->in_build_stack_trace)
+        return;
+    rt->in_build_stack_trace = true;
+
+    // Save exception because conversion to double may fail.
+    saved_exception = JS_GetException(ctx);
+
+    // Extract stack trace limit.
+    d_stack_trace_limit = 10;
+    JS_ToFloat64(ctx, &d_stack_trace_limit, ctx->error_stack_trace_limit);
+    if (isfinite(d_stack_trace_limit))
+        stack_trace_limit = d_stack_trace_limit;
```

Comment:
UB when `d_stack_trace_limit < INT32_MIN || d_stack_trace_limit > INT32_MAX` or when it's negative zero.

Reply:
Oh. I'll look into that!